ReactNativeClient/lib/BaseApplication.js

```diff
@@ -442,6 +442,16 @@ class BaseApplication {
 			refreshNotes = true;
 		}
 
+		if (action.type == 'NOTE_TAG_REMOVE') {
+			if (newState.notesParentType === 'Tag' && newState.selectedTagId === action.item.id) {
+				if (newState.notes.length === newState.selectedNoteIds.length) {
+					this.refreshCurrentFolder();
```

Comment:
This is asynchronous, so it would be best to add `await` here to avoid race conditions.

Reply:
Yes, good point. Fixed.